ansible/post/files/approve_and_issue.sh

```diff
@@ -16,97 +16,97 @@ POWER_COUNT="${2}"
 # Var: ${self.triggers.approve}
 POWER_PREFIX="${3}"
 
-APPROVED_WORKERS=0
-ISSUED_WORKERS=0
+# Machine Prefix
+MACHINE_PREFIX="${POWER_PREFIX}-worker"
+
+# Check count for Power worker/s to be added is not zero
+if [ "0" -eq "${POWER_COUNT}" ]
+then
+  echo "Supplied Worker count is zero hence exiting."
+  exit 0
+fi
 
+# Setting values for variables
+APPROVED_WORKERS=0
 IDX=0
-while [ "$IDX" -lt "121" ]
+READY_COUNT=$(oc get nodes -l kubernetes.io/arch=ppc64le | grep "${MACHINE_PREFIX}" | grep -v NotReady | grep -c Ready)
+
+# Approce CSR and Check Ready status
+while [ "${READY_COUNT}" -ne "${POWER_COUNT}" ]
 do
+  
   export HTTPS_PROXY="http://${PROXY_SERVER}:3128"
 
-  echo "Try number: ${IDX}"
-  echo "List of Power Workers: "
+  echo "List of All Power Workers: "
   oc get nodes -l 'kubernetes.io/arch=ppc64le' -o json | jq -r '.items[] | .metadata.name'
   echo ""
```

Comment:
```suggestion
  oc get nodes -l 'kubernetes.io/arch=ppc64le''
  echo ""
```

Reply:
I think less is more here.

Reply:
My thought was to list all Power workers first and then list Power workers to be added with Prefix. [Because it could be possible that some power workers are already added to OCP cluster by some other means]

Reply:
makes sense